pkg/handlers/clusterlogging/projects_service.go

```diff
@@ -0,0 +1,50 @@
+package clusterlogging
+
+import (
+	"time"
+
+	"github.com/bluele/gcache"
+	"github.com/openshift/elasticsearch-proxy/pkg/clients"
+	"github.com/openshift/elasticsearch-proxy/pkg/handlers/clusterlogging/types"
+	log "github.com/sirupsen/logrus"
+)
+
+type projectsService struct {
+	cache gcache.Cache
+}
+
+func NewRolesService(size int, expiry time.Duration, client clients.OpenShiftClient) *projectsService {
```

Comment:
My original thought was to push the project fetch into the same place we fetch username and roles so it would be one stop and a single cache.  We would then need to pass through via the context so other handlers could do with it whatever they required.  ClusterLogging would seed its permissions using them.  Jaeger could seed permissions differently if needed.  Do you have any concerns about their being multiple caches, expiration times, etc

Reply:
I think one cache is might be better from sar/projects synchronization poin of view. I have moved the retrieval of projects to the auth handler.